components/mailbox/src/Mailbox.svelte

```diff
@@ -250,9 +251,12 @@
   async function threadStarred(event: CustomEvent) {
     if (starredThreads.has(event.detail.thread)) {
       starredThreads.delete(event.detail.thread);
+      event.detail.thread.starred = false;
     } else {
       starredThreads.add(event.detail.thread);
+      event.detail.thread.starred = true;
     }
+    updateThreadStatus(event.detail.thread);
```

Comment:
Are there any un-intended consequences of not awiting the network request? 🤔

Reply:
I don't think it would affect the behaviour here in this case, But you're right, adding an await is a good idea to ensure the update is complete. Thank you!